### DISCUSSION TITLE
Combine mock and limit-count plugin

### DISCUSSION THREAD

**Discussion by xtbe** [apache/apisix#10280] [Q&A]
I was successful in implementing an API with the mock plugin. Also I was able to use the limit-count plugin with an upstream connected. However when I combine the mock and limit-count plugin I am not able to get the rate limit working. 

Can these two be combined or is that not possible? Or is there some automatic limit in place for mocks?

If not, I do believe it would be a good idea to be able to put limits as this would avoid getting the gateway in a bad shape due to e.g mock requests overload

Thanks already for any feedback

/Tom

**Comment by Zeeeeta** [comment]
Same thought when I want to use mock API to demo some feature like rate limiting and ip-restriction

**Comment by MekelCon** [comment]
Hello
It is realated **to plugin default execution order** please see : [plugins-execution-order](https://apisix.apache.org/docs/apisix/terminology/plugin/#plugins-execution-order)
In the default config : 
```
#...
plugins:                           # plugin list (sorted by priority)
...
  - mocking                        # priority: 10900
...
  - limit-count                    # priority: 1002
....
```
It means _mocking_ plugin will be executed BEFORE _limit-count_ plugin.
You can override the priority of plugin using the _meta field see : [Plugin configuration](https://apisix.apache.org/docs/apisix/terminology/plugin/#plugin-common-configuration)

**Reply by xtbe** [reply]
Thank you for the info and educating me on this. Will give it a try.

Thanks for the help!

/Tom
